### ISSUE TITLE
Jest Coverage Not Working

### ISSUE THREAD

**Issue by blink1073** [jupyterlab/jupyterlab#6415]
**Describe the bug**
Coverage reports are not being generated.

**To Reproduce**

1. Go to 'tests/test-application'
2. Run `jlpm coverage`
3. Note the errors of the form:

```
2019-05-24T07:57:40.0336829Z @jupyterlab/test-application: ERROR: ..\..\packages\application\src\index.ts: Emit skipped
2019-05-24T07:57:40.0337497Z @jupyterlab/test-application: STACK: TypeError: ..\..\packages\application\src\index.ts: Emit skipped
2019-05-24T07:57:40.0337992Z @jupyterlab/test-application:     at getOutput (D:\a\1\s\node_modules\ts-jest\dist\compiler.js:160:23)
2019-05-24T07:57:40.0338547Z @jupyterlab/test-application:     at Object.compile (D:\a\1\s\node_modules\ts-jest\dist\compiler.js:206:25)
2019-05-24T07:57:40.0339383Z @jupyterlab/test-application:     at TsJestTransformer.process (D:\a\1\s\node_modules\ts-jest\dist\ts-jest-transformer.js:101:41)
2019-05-24T07:57:40.0340180Z @jupyterlab/test-application:     at ScriptTransformer.transformSource (D:\a\1\s\node_modules\@jest\transform\build\ScriptTransformer.js:387:35)
2019-05-24T07:57:40.0340733Z @jupyterlab/test-application:     at _default (D:\a\1\s\node_modules\@jest\reporters\build\generateEmptyCoverage.js:57:9)
2019-05-24T07:57:40.0341199Z @jupyterlab/test-application:     at Object.worker (D:\a\1\s\node_modules\@jest\reporters\build\coverage_worker.js:80:45)
2019-05-24T07:57:40.0341669Z @jupyterlab/test-application:     at D:\a\1\s\node_modules\@jest\reporters\build\coverage_reporter.js:326:45
2019-05-24T07:57:40.0342105Z @jupyterlab/test-application:     at Generator.next (<anonymous>)
2019-05-24T07:57:40.0342566Z @jupyterlab/test-application:     at asyncGeneratorStep (D:\a\1\s\node_modules\@jest\reporters\build\coverage_reporter.js:124:24)
2019-05-24T07:57:40.0343013Z @jupyterlab/test-application:     at _next (D:\a\1\s\node_modules\@jest\reporters\build\coverage_reporter.js:144:9)
```

**Comment by blink1073** [comment]
The obscured message from TS was:

```
"File '/home/silvester/workspace/jupyter/lab/packages/application/src/tokens.ts' is not under 'rootDir' '/home/silvester/workspace/jupyter/lab/tests/test-application/src'. 'rootDir' is expected to contain all source files."
```

I think the right solution here is to move our test files to live in the same file structure as our source files and import from relative paths in the tests, following this [example](https://github.com/mtiller/ts-jest-sample).

There is nothing tying us to the existing structure.  We can continue to use `extends` for tsconfig, project build for TS, and the name of the package for Lerna targeting.  The only substantive change would be to put the karma config in testutils so we can import it easily.

**Comment by blink1073** [comment]
cc @saulshanabrook, @jasongrout, any objections?

**Comment by saulshanabrook** [comment]
@blink1073 That sounds good to me. I am not particularly invested in our current structure.

**Comment by jasongrout** [comment]
Interesting. We originally had the test files in each package, then you moved them out to their own parallel directory structure. Do you remember the reasons for doing that?

**Comment by jasongrout** [comment]
Why is coverage complaining about this, but we can indeed still compile in other ways? It sounds like the coverage plugin is not support compiling the way we are normally compiling. Is there an option in the coverage plugin we can use?

**Comment by blink1073** [comment]
We had consolidated to reuse components from a parent directory, before we had testutils. Coverage is something that is built in to jest, but it rubbing against TS config capabilities because of our directory structure.  I'd rather go with the established and supported pattern. 😀

**Comment by blink1073** [comment]
Fixed in the process of fixing #5173
